src/coreclr/jit/gentree.cpp

```diff
@@ -17652,6 +17658,197 @@ bool GenTreeVecCon::HandleArgForHWIntrinsicCreate(GenTree* arg, int argIdx, simd
 
     return false;
 }
+
+// TODO-XArch-AVX512: Keep only one implementation once GenTreeVecCon supports gtSimd64Val.
+#if defined(TARGET_XARCH)
+//----------------------------------------------------------------------------------------------
+// IsHWIntrinsicCreateConstant: Determines if a HWIntrinsic node represents a vector constant
+//
+//  Arguments:
+//     node      - The node to check
+//     simd64Val - The vector constant being constructed
+//
+//  Returns:
+//     true if node represents a constant; otherwise, false
+bool GenTreeVecCon::IsHWIntrinsicCreateConstant(GenTreeHWIntrinsic* node, simd64_t& simd64Val)
```

Comment:
Same here. If we made this templated or took a `void* simdVal, uint32_t simdSize` (and then asserted `simdSize <= node->GetSimdSize()`), then we could share the code more easily with the `simd32_t` path and avoid the duplication.